SqlAgDatabase: Adding statementTimeout and Replace for restore

Teutenberg opened the pull request:
#### Pull Request (PR) description
"SqlServerDscHelper.psm1": I've added a new parameter "StatementTimeout" to function "Connect-SQL", and "Invoke-Query". 

"MSFT_SqlAGDatabase.psm1": I've added -StatementTimeout 0 on the long running commands.

#### This Pull Request (PR) fixes the following issues
- Should fix #1358

#### Task list
- [x] Added an entry under the Unreleased section of the change log in the CHANGELOG.md.
      Entry should say what was changed, and how that affects users (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [x] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/sqlserverdsc/1362)
<!-- Reviewable:end -->

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/PowerShell/SqlServerDsc?pullRequest=1362) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/PowerShell/SqlServerDsc?pullRequest=1362) before we can accept your contribution.<br/><br/>:x: Teutenberg [sign now](https://cla.opensource.microsoft.com/PowerShell/SqlServerDsc?pullRequest=1362)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/PowerShell/SqlServerDsc?pullRequest=1362) it.</sub>

Teutenberg commented:
Updated CHANGELOG.md
I will take a look at the unit tests, I'm guessing it's the ones for SqlServerDscHelper as I added a new parameter to two functions.

Teutenberg commented:
I fix the unit test and all is green now.

johlju commented:
@Teutenberg Thanks for this PR! Due to a recent merge there are merge conflicts since your PR is not based on the latest changes. Could you please rebase against branch dev using `git rebase` (not by using `git pull` or `git merge`, to keep the commit history). If you don't know how to rebase your local dev and working branch, please look at how to [Resolve merge conflicts](https://github.com/PowerShell/DscResources/blob/master/GettingStartedWithGitHub.md#resolve-merge-conflicts).
Let me know if you need any assistance.
